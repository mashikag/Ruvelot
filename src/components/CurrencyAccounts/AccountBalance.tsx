import React, { ReactElement } from 'react'
import { useSelector } from 'react-redux'
import styled from 'styled-components'
import { StoreState } from '../../store/types'

function CurrencyAccountBalance(): ReactElement {
  const balance: number = useSelector(
    ({
      currencyAccounts: {
        currencyAccountBalances,
        selectedCurrencyAccount
      }
    }: StoreState) =>
      currencyAccountBalances[selectedCurrencyAccount] || 0
  )

  const balanceWithTwoDecimalPoints = balance.toFixed(2);

  return (
    <AccountBalance title={balanceWithTwoDecimalPoints}>
      {balanceWithTwoDecimalPoints}
    </AccountBalance>
  )
}

export default CurrencyAccountBalance

const AccountBalance =
  styled.span({
    fontSize: '2em',
    lineHeight: '2.1em',
    width: '200px',
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
  })