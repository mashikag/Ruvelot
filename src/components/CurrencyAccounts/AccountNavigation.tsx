import React from 'react'

const AccountNavigation: React.SFC<{}> =
  () => <div>Account Navigation</div>

export default AccountNavigation
