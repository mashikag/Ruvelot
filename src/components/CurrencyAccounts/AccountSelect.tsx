import React, { ReactElement, useCallback } from 'react'
import { useSelector, useDispatch, shallowEqual } from 'react-redux'
import { Currency, StoreState } from '../../store/types';
import { selectCurrencyAccount } from '../../store/index';
import CurrencySelect from '../CurrencySelect/CurrencySelect';

function AccountSelect(): ReactElement<any> {
  const dispatch = useDispatch();

  const currencyAccounts = useSelector<StoreState, Currency[]>(
    ({ currencyAccounts }) =>
      (Object.keys(currencyAccounts.currencyAccountBalances)) as Currency[],
    shallowEqual
  )

  const onAccountSelection = useCallback(
    (accountCurrency: Currency) =>
      dispatch(selectCurrencyAccount(accountCurrency)),
    [dispatch]
  )

  return <CurrencySelect currencies={currencyAccounts} onCurrencyChange={onAccountSelection} />
}

export default AccountSelect