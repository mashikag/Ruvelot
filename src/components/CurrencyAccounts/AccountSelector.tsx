import React, { ReactElement, ChangeEventHandler } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { Currency, StoreState } from '../../store/types';
import { selectCurrencyAccount } from '../../store/index';

function getAccountSelectionItems(currencyAccounts: Currency[]):
  JSX.Element[] {
  return currencyAccounts.map(
    currencyAccount =>
      <option key={currencyAccount} value={currencyAccount}>{currencyAccount}</option>
  )
}

function AccountSelector(): ReactElement<any> {
  const dispatch = useDispatch();

  const currencyAccounts = useSelector<StoreState, Currency[]>(
    ({ currencyAccounts }) =>
      (Object.keys(currencyAccounts.currencyAccountBalances)) as Currency[]
  )

  const onAccountSelection: ChangeEventHandler<HTMLSelectElement> = ({ target }) => {
    const accountCurrency = target.value as Currency
    dispatch(selectCurrencyAccount(accountCurrency))
  }

  return (
    <select onChange={onAccountSelection}>
      {getAccountSelectionItems(currencyAccounts)}
    </select>
  );
}

export default AccountSelector
