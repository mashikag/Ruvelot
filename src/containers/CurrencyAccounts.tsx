import React from 'react';
import { CurrencyAccountsState } from '../store/types';

class CurrencyAccountsContainer
  extends React.Component<CurrencyAccountsState> {

  render(): React.ReactElement<CurrencyAccountsState> {
    return (<div></div>);
  }
}

export default CurrencyAccountsContainer;
