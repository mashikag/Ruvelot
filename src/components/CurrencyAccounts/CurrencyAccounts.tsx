import React, { HTMLAttributes, ReactElement } from 'react'
import styled from 'styled-components'
import AccountBalance from './AccountBalance'
import AccountSelect from './AccountSelect'

function CurrencyAccounts(): ReactElement<any> {
  return (
    <AccountBalanceContainer>
      <CurrencyAccountContainer>
        <AccountBalance />
        <AccountSelect />
      </CurrencyAccountContainer>
      <AccountHeader />
    </AccountBalanceContainer>
  )
}

export default CurrencyAccounts


const AccountHeader =
  styled.header<HTMLAttributes<HTMLElement>>({
    color: 'gray',
    fontSize: '1.1em',
    lineHeight: '1.8em',
    ":after": {
      content: '"Account Balance"'
    }
  })

const AccountBalanceContainer =
  styled.div<HTMLAttributes<HTMLDivElement>>({
    width: '100%',
    textAlign: 'center',
    marginBottom: '20px'
  })

const CurrencyAccountContainer =
  styled.div<HTMLAttributes<HTMLDivElement>>({
    display: 'flux',
    fluxFlow: 'row nowrap',
    justifyContent: 'center'
  })
