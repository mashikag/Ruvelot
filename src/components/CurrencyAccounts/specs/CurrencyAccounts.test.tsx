import React from 'react'
import { shallow } from 'enzyme'
import CurrencyAccounts from '../CurrencyAccounts'

describe('Account Select Component', () => {
  it(`shallow snapshot`, () => {
    const component = shallow(<CurrencyAccounts />)
    expect(component).toMatchSnapshot()
  })
})