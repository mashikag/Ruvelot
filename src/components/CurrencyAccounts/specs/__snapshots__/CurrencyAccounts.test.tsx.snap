// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Select Component shallow snapshot 1`] = `
<styled.div>
  <styled.div>
    <CurrencyAccountBalance />
    <AccountSelect />
  </styled.div>
  <styled.header />
</styled.div>
`;
